pkg/storage/endpoint/tso_keyspace_group.go

```diff
@@ -84,6 +88,30 @@ type KeyspaceGroup struct {
 	KeyspaceLookupTable map[uint32]struct{} `json:"-"`
 }
 
+// InSplit checks if the keyspace group is in split state.
+func (kg *KeyspaceGroup) InSplit() bool {
+	return kg != nil && kg.SplitState != nil
+}
+
+// IsSplitTo checks if the keyspace group is in split state and is the split-to keyspace group.
+func (kg *KeyspaceGroup) IsSplitTo() bool {
+	return kg.InSplit() && kg.SplitState.SplitFrom != kg.ID
+}
+
+// IsSplitFrom checks if the keyspace group is in split state and is the split-from keyspace group.
+func (kg *KeyspaceGroup) IsSplitFrom() bool {
+	return kg.InSplit() && kg.SplitState.SplitFrom == kg.ID
+}
+
+// SplitFrom returns the split-from keyspace group ID. When the keyspace group is the split-from group
+// itself, it will return its own ID.
+func (kg *KeyspaceGroup) SplitFrom() uint32 {
+	if kg.InSplit() {
+		return kg.SplitState.SplitFrom
+	}
+	return 0
```

Comment:
Currently, we check the split status before every `SplitSource()` call in all logic, so there won't be any problems. However, you are right that to avoid potential bugs in the future, I should modify this part.